### PR TITLE
fix: update dimension consts

### PR DESCRIPTION
**Pull request by filipgutica**
# Summary

Further categorize dimension based on what their intended behaviour in filters will be. 

Request dimensions should be broken down into 3 categories: 
- Fetchable dimensions. These are dimensions that require a fetch to the backend in order to populate filter values and only from those items can filters be created. User added values are not permitted.
- Searchable local dimensions. These are dimensions for which values are populated by a local list and locally searchable. User added values are permitted.
- Text based dimensions. These require a free form input for which the user will input a filter value, or an empty multiselect which users are to add values to.

Explore dimensions are broken down to 
- Fetchable dimensions. These are dimensions that require a fetch to the backend in order to populate filter values and only from those items can filters be created. User added values are not permitted.
- Searchable local dimensions. These are dimensions for which values are populated by a local list and locally searchable. User added values are permitted.

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts [hunk 1]
```diff
@@ -28,18 +28,78 @@ export const queryableRequestDimensions = [
...
+  'data_plane_node',
+  'gateway_service',
+  'route',
+] as const satisfies QueryableRequestDimensions[]
+
+export type FetchableRequestDimensions = typeof fetchableRequestDimensions[number]
+
+// These are dimensions that can be searched locally (i.e. in-memory) in the frontend
+// Will have a pre-populated list that can be searched from defined in the host app
+export const searchableLocalRequestDimensions = [
```
Comment (alxgrant) [inline packages/analytics/analytics-utilities/src/types/explore/requests.ts:60]: I think we will include `response_source` in this list

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts [hunk 2]
```diff
@@ -28,18 +28,78 @@ export const queryableRequestDimensions = [
   'service_port',
   'service_protocol',
   'sse',
+  'status_code',
```
Comment (alxgrant) [inline packages/analytics/analytics-utilities/src/types/explore/requests.ts:31]: What does including `status_code` here help with? We expect `status_code` to be a number value in the API, but now it's being included in ways that expect a string filter value like the `RequestEqualsFilter`.

**Review by adorack — CHANGES_REQUESTED** [review]
I'd like to review (once I get caught up).

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts (hunk 2 above)
Comment (alxgrant) [reply packages/analytics/analytics-utilities/src/types/explore/requests.ts:31]: Ah nvm I see you had a comment in slack about this and I see now how the dimensions are used below with the local searchable list.

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts (hunk 1 above)
Comment (filipgutica) [reply packages/analytics/analytics-utilities/src/types/explore/requests.ts:60]: Are we sure that it's a searchable one, not a text based one, is it something that will have a pre-populated list to select from?

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts (hunk 2 above)
Comment (filipgutica) [reply packages/analytics/analytics-utilities/src/types/explore/requests.ts:31]: It behaves more like a dimension filter in that there's a searchable list of things to select from locally when filtering, in addition to allowing user inputted values. I don't think there's another metric that behaves like this.

I may be wrong, in that case, we should split the metrics into two categories, one for metrics for which we pre-populate a list of things the user can pick from, and ones for which we do not

File: packages/analytics/analytics-utilities/src/types/explore/requests.ts (hunk 1 above)
Comment (alxgrant) [reply packages/analytics/analytics-utilities/src/types/explore/requests.ts:58]: Yea, we would like to show "kong" and "upstream". It's in the first sheet here https://docs.google.com/spreadsheets/d/1XQW4sXPlR7_7kUKxhKGTNueeQUZF3Pr4FJnbRz00axE/edit?gid=0#gid=0

**Pull request closed without merging**